shared/print.go

```diff
@@ -0,0 +1,71 @@
+package shared
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+var Flags struct {
+	Json bool
+}
+
+func PrintProgress(msg string) {
+	if !Flags.Json {
+		fmt.Printf("* %s\n", msg)
+	}
+}
+
+func PrintItem(msg string) {
+	if !Flags.Json {
+		fmt.Printf("  - %s\n", msg)
+	}
+}
+
+func PrintIndented(v any) {
+	b, _ := json.MarshalIndent(v, "", "  ")
+	fmt.Println(string(b))
+}
+
+func PrintResult(result any, key string, message string) {
+	if Flags.Json {
+		PrintIndented(map[string]any{"ok": true, key: result})
+	} else {
+		b, _ := json.Marshal(result)
+		fmt.Printf("* %s: %s\n", message, string(b))
+	}
+	os.Exit(0)
+}
+
+func PrintResults[T any](results []T, key, noun, verb, singular, plural string) {
+	if Flags.Json {
+		PrintIndented(map[string]any{"ok": true, "count": len(results), key: results})
+	} else {
+		if len(results) == 1 {
+			fmt.Printf("* %s 1 %s\n", verb, singular)
+		} else {
+			fmt.Printf("* %s %d %s\n", verb, len(results), plural)
+		}
+		for i, result := range results {
+			b, _ := json.Marshal(result)
+			fmt.Printf("  - %s %d: %s\n", noun, i, string(b))
+		}
+	}
+	os.Exit(0)
```

Comment:
Do we always want to exit after print? 
I don't like adding exit in inner function, make code harder to read